src/features/diagnosticsProvider.ts

```diff
@@ -234,24 +234,26 @@ class DiagnosticsProvider extends AbstractSupport {
 
         let source = new vscode.CancellationTokenSource();
         let handle = setTimeout(() => {
-            serverUtils.codeCheck(this._server, { FileName: document.fileName }, source.token).then(value => {
+            if (this._server.isRunning()) {
```

Comment:
I realised that the makeRequest function to the server already checks if the server is running, so that would cover all the scenarios here. The main problem was that we were assuming that the server has been started even when it was not.

Reply:
Added error handling when the rejected requests are received.